src/annoylib.h

```diff
@@ -457,7 +513,14 @@ template<typename S, typename T, typename Distance>
           children.push_back(n);
       }
 
-      Distance::create_split(children, _f, &_random, m);
+      {
+        // Sample two random points from the set of nodes so that i != j
```

Comment:
It makes it easier to implement a random number generator. The sampling of two points seems to be done only at this place so why not have the code here?